### DISCUSSION TITLE
價格波動方向驗證（期貨、指數)

### DISCUSSION THREAD

**Discussion by Will150143** [ChiuAlfredo/options_day_data#23] [研究]
# 研究題目
1.不同價性的選擇權分報酬率和期貨分報酬率的相關性
2.不同價性的選擇權分報酬率和加權指數分報酬率的相關性
3.不同價性的選擇權分報酬率在開盤時是否與期貨脫鉤
## 研究方法
題目1.2. 不同價性的選擇權的分報酬率對期貨和加權指數分報酬率分別做皮爾森相關分析，得出相關係數來比較
題目3. 取開盤後10分鐘不同價性的選擇權的分報酬率對期貨和加權指數分報酬率做皮爾森相關分析，得出相關係數並與題目1.2.的結果比較
## 使用資料
價內、價平、價外買和賣權分報酬率、期貨和加權指數分報酬率，分報酬率計算方式為：Ln(當分鐘成交價)-Ln(前一分鐘成交價)

以結算日開盤價做為價性的標準
### 買權（call option）
$\frac{K}{S}$​ 範圍 | 價性分類
-- | --
$0.98 \leq \frac{K}{S} ​<0.99 $ | 價內 (ITM)
$0.99\leq \frac{K}{S} <1.01 $ | 價平 (ATM)
$1.01\leq \frac{K}{S} ​<1.02 $ | 價外 (OTM)

### 賣權（put option）
$\frac{K}{S}$​ 範圍 | 價性分類
-- | --
$1.01\leq \frac{K}{S} ​<1.02 $ | 價內 (ITM)
$0.99\leq \frac{K}{S} <1.01 $ | 價平 (ATM)
$0.98 \leq \frac{K}{S} ​<0.99 $ | 價外 (OTM)

## 預期結果
- 不同價性的選擇權分報酬率對期貨和指數報酬率低度相關（|r|<0.3)、中度相關(0.3≤|r|<0.6)、高度相關(0.6≤|r|<0.9)、完全相關(|r|=1)
- 剛開盤時不同價性的選擇權分報酬率對期貨和指數報酬率低度相關（脫鉤)、顯著相關(可能脫鉤)、高度相關(沒有脫鉤)
### 補充資料
- 皮爾森相關分析
  - 介紹
皮爾森相關係數(r)的變化範圍為-1到1。值為1意味著所有的點都落在一條直線上，且 Y 隨著 X 的增加而增加。值為−1意味著所有的點都落在一條直線上，且 Y 隨著 X 的增加而減少。值為0意味著兩個變量之間沒有線性關係。 
  - 公式
![image](https://github.com/ChiuAlfredo/options_day_data/assets/133984319/bea2fd23-48fa-409e-9aad-4a8d12caee3a)
  - 判斷依據
按四級劃分：|r|<0.3為低度相關；0.3≤|r|<0.6為中度相關；0.6≤|r|<0.9為高度相關，|r|=1為完全相關。
依據來源：https://myweb.scu.edu.tw/~swlean/statistics4.htm
- log return
  - 公式和說明
![image](https://github.com/user-attachments/assets/82573f69-9b25-4e3d-a882-1f493bb5278e)
## 筆記
1.資料中每一格成交價都是那一分鐘的加權平均價
2.倘若某段時間沒有交易，那這一段時間呈現的價格會是最近一次有交易產生時的那一分鐘的價格
這會導致兩件事情：
- 剛開盤沒有交易就會導致那段時間無法計算報酬率，因為ln(0)是無法計算的，該儲存格會是空值
- 開盤以外沒有交易的時間會出現報酬率為0的狀況，因為前一分鐘和當分鐘的價格是一樣的
因應這個狀況我目前採取的做法是去掉空值來計算平均去平滑化資料

**Comment by ChiuAlfredo** [comment]
1. 可以把要分組做的實驗寫出來，例如1.選擇權跟指數 2. 選擇權跟期貨
2. 不切割時間要怎麼去做
3. 是要選擇權所有資料做比較比較還是只有結算日
4. (P2-P1)/P1這個應該是波動率吧，但是你上面說的是波動
![image](https://github.com/ChiuAlfredo/options_day_data/assets/91011574/f0425360-6ab4-492a-8572-6a4f8c3432f0)
你這裡寫波動，但是公式應該是波動率

6. 可以再多說明是什麼相關係數，可能是pearson相關係數還是啥
7. 算出相關係數後還可以怎麼去做比較，可能使用anova 去看到底有沒有顯著的差異，或是沒有
8. 可能要說明價性以及買賣權目前都沒有說明

**Reply by ChiuAlfredo** [reply]
1.OK，我會補完整一點
2.我原本是想比整天的，但想了一下還是切好了
3.我依舊看不懂全部有交易是什麼意思，但我只做結算日當天
4.我原以為是要用變化率當資料，還是說你要的是標準差，但查資料後發現標準差也叫波動率（標準差無法反映波動方向)，所以我不確定你指的波動是什麼
5.沒問題，我會補上
6.可以，但做ANOVA會做到期貨跟指數一組的，我會分開做F檢定
7.沒問題，我會補上

**Comment by ChiuAlfredo** [comment]
所以公式上x會是選擇權價格，y是指數或是期貨嗎

**Reply by Will150143** [reply]
為什麼會有x跟y 我沒有要跑迴歸阿 雖然要也是可以

**Reply by ChiuAlfredo** [reply]
你的公式裡有x跟y

**Reply by Will150143** [reply]
我想是可以互換的 但統一x是選擇權價格好了

**Comment by ChiuAlfredo** [comment]
波動跟波動率應該不太一樣可能要注意用詞

**Reply by Will150143** [reply]
Ok 我會注意

**Comment by TsuYiii** [comment]
1.結算日當天的價平（0.99<S/K<1.01)買、賣權 時間單位為1分鐘，K/S
2.變化率的公式指的是前一秒及當秒嗎？還是是當秒與下一秒呢？

**Reply by Will150143** [reply]
前一分鐘跟當一分鐘 應該沒有秒資料

**Comment by Will150143** [comment]
更新進度
目前已完成資料篩選和計算
計算方式更改為：當分鐘價格/前一分鐘價格取對數（log return)
接著C、P、加權和期貨各自取平均（四個項目都會得出一個平均值)，再分時段進行相關係數的計算

**Comment by ChiuAlfredo** [comment]
我以為是在這段是在驗證說，剛開盤選擇權價格跟指數價格脫鉤的問題

**Reply by Will150143** [reply]
我會把這個當作延伸來做
目前的狀況會是先做完整天的相關係數，接著再計算開盤時的兩者的相關係數來比高低

**Comment by Will150143** [comment]
## 9:01-13:30
### 買權（call option）
#### 樣本數
| 價性  | 數量 |
| :---         |     :---:      |
| 價內 (ITM)   | 各4304     |
| 價平 (ATM)     | 各4304       |
| 價外 (OTM)     | 各4304       |

| 價性 | 對加權相關係數 | 對期貨相關係數 |
| :---         |     :---:      |          ---: |
| 價內 (ITM)   | 0.3224     | 0.5922    |
| 價平 (ATM)     | 0.2461       | 0.5901      |
| 價外 (OTM)     | 0.1221       | 0.2703      |

### 賣權（put option）
#### 樣本數
| 價性  | 數量 |
| :---         |     :---:      |
| 價內 (ITM)   | 各4304     |
| 價平 (ATM)     | 各4304       |
| 價外 (OTM)     | 各4304       |

| 價性 | 對加權相關係數 | 對期貨相關係數 |
| :---         |     :---:      |          ---: |
| 價內 (ITM)   | -0.3417     | -0.5997    |
| 價平 (ATM)     | -0.2476       | -0.6346      |
| 價外 (OTM)     | -0.1932       | -0.4222      |
## 9:01-9:10
### 買權（call option）
#### 樣本數
| 價性  | 數量 |
| :---         |     :---:      |
| 價內 (ITM)   | 各144     |
| 價平 (ATM)     | 各144       |
| 價外 (OTM)     | 各144       |

| 價性 | 對加權相關係數 | 對期貨相關係數 |
| :---         |     :---:      |          ---: |
| 價內 (ITM)   | 0.3885     | 0.7060    |
| 價平 (ATM)     | 0.3325       | 0.8906      |
| 價外 (OTM)     | 0.0673       | 0.7468      |

### 賣權（put option）
#### 樣本數
| 價性  | 數量 |
| :---         |     :---:      |
| 價內 (ITM)   | 各144     |
| 價平 (ATM)     | 各144       |
| 價外 (OTM)     | 各144       |

| 價性 | 對加權相關係數 | 對期貨相關係數 |
| :---         |     :---:      |          ---: |
| 價內 (ITM)   | -0.3132     | -0.6113    |
| 價平 (ATM)     | -0.2031       | -0.9021      |
| 價外 (OTM)     | -0.1924       | -0.7399      |
## 9:11-9:30
### 買權（call option）
#### 樣本數
| 價性  | 數量 |
| :---         |     :---:      |
| 價內 (ITM)   | 各320     |
| 價平 (ATM)     | 各320       |
| 價外 (OTM)     | 各320       |

| 價性 | 對加權相關係數 | 對期貨相關係數 |
| :---         |     :---:      |          ---: |
| 價內 (ITM)   | 0.599     | 0.7078    |
| 價平 (ATM)     | 0.6425       | 0.8237      |
| 價外 (OTM)     | 0.5195       | 0.6199      |

### 賣權（put option）
#### 樣本數
| 價性  | 數量 |
| :---         |     :---:      |
| 價內 (ITM)   | 各320     |
| 價平 (ATM)     | 各320       |
| 價外 (OTM)     | 各320       |

| 價性 | 對加權相關係數 | 對期貨相關係數 |
| :---         |     :---:      |          ---: |
| 價內 (ITM)   | -0.5851     | -0.657    |
| 價平 (ATM)     | -0.6123       | -0.8061      |
| 價外 (OTM)     | -0.4835       | -0.6521      |

**Reply by Will150143** [reply]
## 小結
- 選擇權對期貨的相關係數較對加權的相關係數高→選擇權對期貨的連動程度較高
- 價內與價平的選擇權對期貨與加權的相關係數比價外的選擇權高→價內和價平的選擇權對兩指數的連動程度較價外高
- 開盤10分鐘跟後20分鐘，選擇權對期貨的相關係數沒有明顯變化，但對加權的相關係數變高→剛開盤時選擇權和加權的連動程度低，但之後連動程度會提高

**Comment by TsuYiii** [comment]
請問在預期結果中，無相關、低相關、高相關與皮爾森公式中的相關程度相同區間嗎？如相同是，是否該以同樣名稱取名標示呢？
